app/perfsymboltable.cpp

```diff
@@ -473,33 +473,61 @@ static QFileInfo findDebugInfoFile(const QString &root, const QString &file,
 
     QFileInfo debugLinkFile;
 
+    // try in .debug folder
     if (!folder.isEmpty()) {
-        debugLinkFile.setFile(dir, folder + QDir::separator() + debugLinkString);
+        debugLinkFile.setFile(dir.path() + QDir::separator() + folder + QDir::separator() + QLatin1String(".debug")
+                              + QDir::separator() + debugLinkString);
         if (debugLinkFile.isFile())
             return debugLinkFile;
     }
 
-    debugLinkFile.setFile(dir, file + QDir::separator() + debugLinkString);
+    debugLinkFile.setFile(dir.path() + QDir::separator() + file + QDir::separator() + QLatin1String(".debug")
+                          + QDir::separator() + debugLinkString);
     if (debugLinkFile.isFile())
         return debugLinkFile;
 
-    // try again in .debug folder
-    if (!folder.isEmpty()) {
-        debugLinkFile.setFile(dir, folder + QDir::separator() + QLatin1String(".debug")
-                                    + QDir::separator() + debugLinkString);
-        if (debugLinkFile.isFile())
+    // try again in /usr/lib/debug folder
+    // some distros use for example /usr/lib/debug/lib (ubuntu) and some use /usr/lib/debug/usr/lib (fedora)
+    const auto usr = QDir::separator() + QLatin1String("usr") + QDir::separator();
+    auto folderWithoutUsr = folder;
+    folderWithoutUsr.replace(usr, QDir::separator());
+
+    // make sure both (/usr/ and /) are searched
+    for (const auto& path : {folderWithoutUsr, usr + folderWithoutUsr}) {
+        debugLinkFile.setFile(dir.path() + QDir::separator() + QLatin1String("usr") + QDir::separator()
+                              + QLatin1String("lib") + QDir::separator() + QLatin1String("debug") + QDir::separator()
+                              + path + QDir::separator() + debugLinkString);
+
+        if (debugLinkFile.isFile()) {
             return debugLinkFile;
+        }
     }
 
-    debugLinkFile.setFile(dir, file + QDir::separator() + QLatin1String(".debug")
-                                + QDir::separator() + debugLinkString);
-    if (debugLinkFile.isFile())
+    debugLinkFile.setFile(dir.path() + QDir::separator() + QLatin1String("usr") + QDir::separator()
+                          + QLatin1String("lib") + QDir::separator() + QLatin1String("debug") + QDir::separator()
```

Comment:
Why all the QDir::separator() business? Not only those paths are very Unix specific, but anyway QFile supports '/' on Windows too.

Reply:
I just wanted to keep the file in the same style

Reply:
@dfaure-kdab I'm with you here, but this was TQC Ulf's choice back then when he ported this to windows - it's pretty mysterious to me too why this is the way it is. I never tested it on windows myself though...